### PR TITLE
Add light functional test and PR review helper script 

### PR DESCRIPTION
**Pull request by viglesiasce**
This PR helps chart maintainers mostly 😄 

The first change is to add very basic functional testing to each chart that is tested for a give PR. The `verify-release.sh` script ensures that if pods were launched by the chart, that they all go to the Running state within the given interval. Charts that have required values often do not launch any pods, in this case we just echo a WARN message. This test is best at detecting bad images or runtime configs.

The second addition in this PR is the `pr-review.sh` script which intends to give chart maintainers an easy way to run and verify the changes in a pr. 

You would run something like:

```
$ export GOOGLE_APPLICATION_CREDENTIALS=service-account.json
$ PULL_NUMBER=318 ./test/pr-review.sh
```
The first command sets the path to Google Cloud Platform service account credentials that can be used to access the test cluster. The second command checks out the desired PR (318) and then runs it through the e2e tests. After the tests are complete, the script will pause for $VERIFICATION_PAUSE  (default 180) seconds so that the tester can manually inspect the deployment for validity.

### CODE REVIEW

**Comment by viglesiasce** [comment]
@k8s-bot e2e test this

**Comment by prydonius** [comment]
This looks really great to me, thanks @viglesiasce. It would be useful, maybe as a future improvement, if the `pr-review` script didn't require the service account, but just used the context in my kubectl - this way I'd be able to test on my AWS cluster easily. The `e2e` script could be modified to mount `~/.kube/config` if a service account isn't provided.

File: test/pr-review.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+if [ -z "${PULL_NUMBER}" ]; then
+  if [ -z "$1" ];then
+    usage
+  fi
+  PR_NUMBER=$1
+fi
+
+if [ -z "${VERIFICATION_PAUSE}" ]; then
```
Comment (prydonius) [inline test/pr-review.sh]: This line gives me the error: `./test/pr-review.sh: line 20: VERIFICATION_PAUSE: unbound variable` so the default doesn't get set.

Comment (viglesiasce) [reply test/pr-review.sh]: Hmm I ran into this in the other script but not this one, will fix the same way.

**Comment by viglesiasce** [comment]
All set with the VERIFICATION_PAUSE fix @prydonius. Will follow up with a PR to ensure you can use local kube contexts.

**Comment by viglesiasce** [comment]
Updated with the ability to use your local kubeconfig.

**Comment by enisoc** [comment]
cc @foxish: This seems relevant to your goal of running tests to ensure new Kubernetes releases don't break published charts.

**Comment by foxish** [comment]
@enisoc, thanks :) @viglesiasce and I had a discussion about this earlier this week.

**Comment by prydonius** [comment]
@viglesiasce looks like some other changes got accidentally included in this?

**Comment by viglesiasce** [comment]
Shoot.

**Comment by viglesiasce** [comment]
@prydonius sanity restored

**Comment by prydonius** [comment]
@viglesiasce I can't seem to get this working without a service account and just using my local kubeconfig. I get:

```
$ PULL_NUMBER=437 ./test/pr-review.sh
...
./test/e2e.sh: line 26: GOOGLE_APPLICATION_CREDENTIALS: unbound variable
```


```
$ GOOGLE_APPLICATION_CREDENTIALS= PULL_NUMBER=437 ./test/pr-review.sh
...
+ docker run -v /Users/adnan/Projects/containers/charts:/src -v :/service-account.json:ro -e GOOGLE_APPLICATION_CREDENTIALS=/service-account.json -e PULL_NUMBER=437 -e BUILD_NUMBER=0 gcr.io/kubernetes-charts-ci/test-image:v1.10 /src/test/changed.sh
docker: Error response from daemon: invalid volume spec ":/service-account.json:ro": invalid volume specification: ':/service-account.json:ro'.
```

```
$ touch /tmp/emptyfile
$ GOOGLE_APPLICATION_CREDENTIALS=/tmp/emptyfile PULL_NUMBER=437 ./test/pr-review.sh
...
ERROR: (gcloud.auth.activate-service-account) Missing required argument [ACCOUNT]: An account is required when using .p12 keys
```

Anyway, let's merge as-is and fix the kubeconfig support in a follow-up PR?

**Pull request merged**